Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCVideoFrameUtilities.h

```diff
@@ -0,0 +1,197 @@
+        transfer = PlatformVideoTransferCharacteristics::SmpteSt4281;
+        break;
+    case webrtc::ColorSpace::TransferID::kARIB_STD_B67:
+        transfer = PlatformVideoTransferCharacteristics::AribStdB67Hlg;
+        break;
+    case webrtc::ColorSpace::TransferID::kIEC61966_2_1:
+    case webrtc::ColorSpace::TransferID::kUnspecified:
+        break;
+    };
+
+    std::optional<PlatformVideoMatrixCoefficients> matrix;
+    switch (webrtcColorSpace->matrix()) {
+    case webrtc::ColorSpace::MatrixID::kRGB:
+        matrix = PlatformVideoMatrixCoefficients::Rgb;
+        break;
+    case webrtc::ColorSpace::MatrixID::kBT709:
+        matrix = PlatformVideoMatrixCoefficients::Bt709;
+        break;
+    case webrtc::ColorSpace::MatrixID::kFCC:
+        matrix = PlatformVideoMatrixCoefficients::Fcc;
+        break;
+    case webrtc::ColorSpace::MatrixID::kBT470BG:
+        matrix = PlatformVideoMatrixCoefficients::Bt470bg;
+        break;
+    case webrtc::ColorSpace::MatrixID::kSMPTE170M:
+        matrix = PlatformVideoMatrixCoefficients::Smpte170m;
+        break;
+    case webrtc::ColorSpace::MatrixID::kSMPTE240M:
+        matrix = PlatformVideoMatrixCoefficients::Smpte240m;
+        break;
+    case webrtc::ColorSpace::MatrixID::kYCOCG:
+        matrix = PlatformVideoMatrixCoefficients::YCgCo;
+        break;
+    case webrtc::ColorSpace::MatrixID::kBT2020_NCL:
+        matrix = PlatformVideoMatrixCoefficients::Bt2020NonconstantLuminance;
+        break;
+    case webrtc::ColorSpace::MatrixID::kBT2020_CL:
+        matrix = PlatformVideoMatrixCoefficients::Bt2020ConstantLuminance;
+        break;
+    case webrtc::ColorSpace::MatrixID::kUnspecified:
+    case webrtc::ColorSpace::MatrixID::kSMPTE2085:
+    case webrtc::ColorSpace::MatrixID::kCDNCLS:
+    case webrtc::ColorSpace::MatrixID::kCDCLS:
+    case webrtc::ColorSpace::MatrixID::kBT2100_ICTCP:
+        break;
+    };
+
+    std::optional<bool> fullRange;
+    switch (webrtcColorSpace->range()) {
+    case webrtc::ColorSpace::RangeID::kLimited:
+        fullRange = false;
+        break;
+    case webrtc::ColorSpace::RangeID::kFull:
+        fullRange = true;
+        break;
+    case webrtc::ColorSpace::RangeID::kInvalid:
+    case webrtc::ColorSpace::RangeID::kDerived:
+        break;
+    };
+
+    return PlatformVideoColorSpace { primaries, transfer, matrix, fullRange };
+}
+
+static inline VideoFrameRotation videoRotationFromLibWebRTCVideoFrame(const webrtc::VideoFrame& frame)
+{
+    switch (frame.rotation()) {
+    case webrtc::VideoRotation::kVideoRotation_0:
+        return VideoFrameRotation::None;
+    case webrtc::VideoRotation::kVideoRotation_90:
+        return VideoFrameRotation::Right;
+    case webrtc::VideoRotation::kVideoRotation_180:
+        return VideoFrameRotation::UpsideDown;
+    case webrtc::VideoRotation::kVideoRotation_270:
+        return VideoFrameRotation::Left;
+    };
+    RELEASE_ASSERT_NOT_REACHED();
```

Comment:
I think just an ASSERT_NOT_REACHED would be enough. I don't think we need to crash the process in all cases for an unexpected value.